### PR TITLE
[Encoding]: Implement multiple EKeys

### PR DESCRIPTION
**Pull request by MaxtorCoder**
- Add support for Multiple EKeys in Encoding
- Maybe find a way to open a specific EKey.

### CODE REVIEW

**Comment by barncastle** [comment]
Thanks for this. I was waiting until this actually drop before supporting it due to some additional complexities.
I'll review it later this afternoon and get this merged in.

**Comment by MaxtorCoder** [comment]
> Thanks for this. I was waiting until this actually dropped before supporting it due to some additional complexities.
> I'll review it later this afternoon and get this merged in.

Amazing! Thanks, this was also the issue with encoding.

File: TACT.Net/SystemFiles/Encoding/EncodingFile.cs [hunk 1]
```diff
@@ -289,14 +291,14 @@ public bool Remove(CASRecord record, TACTRepo tactRepo = null)
...
 
                 // propagate removal
                 if (tactRepo != null)
                 {
-                    tactRepo.IndexContainer?.Remove(entry.EKey);
-                    tactRepo.DownloadFile?.Remove(entry.EKey);
-                    tactRepo.DownloadSizeFile?.Remove(entry.EKey);
+                    entry.EKeys.ForEach(x => tactRepo.IndexContainer?.Remove(x));
+                    entry.EKeys.ForEach(x => tactRepo.DownloadFile?.Remove(x));
+                    entry.EKeys.ForEach(x => tactRepo.DownloadSizeFile?.Remove(x));
```
Comment (barncastle) [inline TACT.Net/SystemFiles/Encoding/EncodingFile.cs:301]: A bit pedantic but can you change this to iterate once please, it just saves on the loops?  e.g.
```c#
entry.EKeys.ForEach(x =>
{
    _EKeyEntries.Remove(x);

   // propagate removal
   if (tactRepo != null) {
      ...
   }
});
```
Do you know if EKeys are now shared across CKeyEntries? If so we may need to change this.

**Review by barncastle — CHANGES_REQUESTED** [review]

File: TACT.Net/SystemFiles/Encoding/EncodingFile.cs (hunk 1 above)
Comment (MaxtorCoder) [reply TACT.Net/SystemFiles/Encoding/EncodingFile.cs:301]: > Do you know if EKeys are now shared across CKeyEntries? If so we may need to change this.

I have no idea, but I will change the suggested code.

**Review by barncastle — APPROVED** [review]

**Pull request merged**